.github/workflows/CI-pr.yml

```diff
@@ -52,7 +54,16 @@ jobs:
 
       - name: Check formatting
         shell: bash
-        run: cargo +nightly fmt --all --check
+        run: |
+          if ! cargo +nightly fmt --all --check; then
+            echo "::notice::Formatting check failed"
+            cargo +nightly fmt
```

Comment:
Instead of running `cargo fmt` twice, do you think it would make sense to always run it and later check for modified files using `git diff`?

Also, you're missing the `--all` flag to `cargo fmt` in the second invocation.